### ISSUE TITLE
GLFW3

### ISSUE THREAD

**Issue by xyproto** [ginkgo/pink-pony#5]
Hi,

I'm trying to package pink-pony for Arch Linux. The official version of GLFW for Arch is 3.1.1.

My attempt to add full GLFW3 support was not completed, but this is what I've got so far:
- `libglfw` should be changed to `glfw3` in `SConstruct`.
- `GLFW/glfw3.h` should be used instead of `GL/glfw.h` in `src/cinquo.hh` and `external/flextGL/flextGL.c`.
- `src/cinquo.hh` should include `GL/glu`
- `src/Config.cc` should include `GLFW/glfw3.h`
- `GLFW_FULLSCREEN` and `GLFW_WINDOW` should not be used

See this document for more information about making the transition from GLFW 2 to 3: http://www.glfw.org/docs/3.0/moving.html

Please support GLFW3.

Best regards,
Alexander F Rødseth

**Comment by ginkgo** [comment]
Hello xyproto,

porting to GLFW3 shouldn't be too much of an issue. flextGL has a backend for GLFW3 now, so we can generate code for that. 
I didn't know pink-pony still uses GLU. We can probably get rid of that dependency altogether.

**Comment by ginkgo** [comment]
PS: There already is an AUR package for pink-pony: https://aur.archlinux.org/packages/pink-pony/
That depends on glfw2, which is also an AUR package.

One thing which sort of kept me from switching to GLFW3 with pink-pony was that it still hasn't been widely adopted by other distros. Those are usually less quick to switch to newer api versions than Arch is.

**Comment by xyproto** [comment]
Thanks for the reply! I somehow missed the AUR package. It's understandable if supporting GLEW3 is not a priority. Cheers!

**Comment by Turulomio** [comment]
Hi, I played this game time ago. I've been trying to compile it in Gentoo, but GLFW-2 it's not supported anymore. Do you have any plans to update to GLFW3?. 

I forked and try to do my best, but I'm not familiar with GLFW3, sorry.

Thanks for the game

**Comment by ginkgo** [comment]
Hi @Turulomio, hmm.. I remember that I started working on this(created a glfw3 branch for it), but it seems like I never committed the local changes. I see if I can get it to compile and push the current state to the glfw3 branch. 
It shouldn't be too hard to switch to glfw3. The main change is that many functions need a glfw window to be passed as an additional parameter.
